lib/src/timing_specifiers.dart

```diff
@@ -0,0 +1,16 @@
+class StartupTimingType {
+  final String name;
+  const StartupTimingType._(this.name);
+
+  static const StartupTimingType firstComponentRender =
```

Comment:
Continuing this unresolved conversation: https://github.com/Workiva/w_module/pull/115#discussion_r208044080

> The problem with just going with one option now and moving on is introducing a public API that we might change once we support other events. That was the main motivation to combining the PR's. I can undo it easily, but we should keep that in mind.

I don't think that removing all but one constant vs leaving them as-is puts us at any more risk for changing the public API.

Reply:
This comment was in regards to having it be a specific 'useful state' method call versus this enumeration approach.

Reply:
It's pretty trivial, if we go this enum route, to have just one vs all of them. I don't see the advantage of splitting them up into multiple tickets.